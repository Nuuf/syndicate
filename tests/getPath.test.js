/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('gets the path of index and one or more attributes found in data', () => {
  const root = syndicate.createRoot();
  const a = syndicate.add(root, syndicate.create({ someData: 'path' }, 'A'));
  const b = syndicate.add(root, syndicate.create({ someData: 'to' }, 'B'), a);
  const c = syndicate.add(root, syndicate.create({ someData: 'this' }, 'C'), b);
  const d = syndicate.add(root, syndicate.create({ someData: 'little' }, 'D'), c);
  const e = syndicate.add(root, syndicate.create({ someData: 'data' }, 'E'), d);
  const path = syndicate.getPath(root, e, ['someData'], true);
  expect(JSON.stringify(path.someData, null, 2)).toMatchSnapshot();
});
