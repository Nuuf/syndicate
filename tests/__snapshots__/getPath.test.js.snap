// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets the path of index and one or more attributes found in data 1`] = `
"[
  \\"path\\",
  \\"to\\",
  \\"this\\",
  \\"little\\",
  \\"data\\"
]"
`;
